graylog2-web-interface/src/views/components/queries/QueryTitleEditModal.jsx

```diff
@@ -0,0 +1,91 @@
+// @flow strict
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Modal, Button } from 'components/graylog';
+import Input from 'components/bootstrap/Input';
+
+import type { TitlesMap } from 'views/stores/TitleTypes';
+
+/**
+ * Component that allows the user to update a QueryTab title.
+ * It takes the active query title as an argument on open and will use it as the draft.
+ * The open action is getting called outside by referencing this component.
+ * The state of this modal can't be handled inside the QueryTitle, due to the react bootstrap tabs key bindings.
+ * E.g. using the arrow keys inside the title input would result in losing the focus.
+ */
+
+type Props = {
+  /** Function to update the query tab tite on save */
```

Comment:
Please remove this comment as well, the function name should (and is in this case) enough.